sdk/go/api_esc_extensions.go

```diff
@@ -0,0 +1,287 @@
+}
+
+func (c *EscClient) ReadEnvironmentProperty(ctx context.Context, org, envName, openEnvID, propPath string) (*Value, any, error) {
+	prop, _, err := c.EscAPI.ReadOpenEnvironmentProperty(ctx, org, envName, openEnvID).Property(propPath).Execute()
+	v := mapValuesPrimitive(prop.Value)
+	return prop, v, err
+}
+
+func (c *EscClient) CreateEnvironment(ctx context.Context, org, envName string) error {
+	_, _, err := c.EscAPI.CreateEnvironment(ctx, org, envName).Execute()
+	return err
+}
+
+func (c *EscClient) UpdateEnvironmentYaml(ctx context.Context, org, envName, yaml string) (*EnvironmentDiagnostics, error) {
+	diags, _, err := c.EscAPI.UpdateEnvironmentYaml(ctx, org, envName).Body(yaml).Execute()
+	return diags, err
+}
+
+func (c *EscClient) UpdateEnvironment(ctx context.Context, org, envName string, env *EnvironmentDefinition) (*EnvironmentDiagnostics, error) {
+	yaml, err := MarshalEnvironmentDefinition(env)
+	if err != nil {
+		return nil, err
+	}
+
+	diags, _, err := c.EscAPI.UpdateEnvironmentYaml(ctx, org, envName).Body(yaml).Execute()
+	return diags, err
+}
+
+func (c *EscClient) DeleteEnvironment(ctx context.Context, org, envName string) error {
+	_, _, err := c.EscAPI.DeleteEnvironment(ctx, org, envName).Execute()
+	return err
+}
+
+func (c *EscClient) CheckEnvironment(ctx context.Context, org string, env *EnvironmentDefinition) (*CheckEnvironment, error) {
+	yaml, err := MarshalEnvironmentDefinition(env)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.CheckEnvironmentYaml(ctx, org, yaml)
+}
+
+func (c *EscClient) CheckEnvironmentYaml(ctx context.Context, org, yaml string) (*CheckEnvironment, error) {
+	check, _, err := c.EscAPI.CheckEnvironmentYaml(ctx, org).Body(yaml).Execute()
+	var genericOpenApiError *GenericOpenAPIError
+	if err != nil && errors.As(err, &genericOpenApiError) {
+		model := genericOpenApiError.Model().(CheckEnvironment)
+		return &model, err
+	}
+
+	return check, err
+}
+
+func (c *EscClient) DecryptEnvironment(ctx context.Context, org, envName string) (*EnvironmentDefinition, string, error) {
+	env, resp, err := c.EscAPI.DecryptEnvironment(ctx, org, envName).Execute()
+
+	body, bodyErr := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, "", bodyErr
+	}
+
+	return env, string(body), err
+}
+
+func MarshalEnvironmentDefinition(env *EnvironmentDefinition) (string, error) {
+	var bs []byte
+	bs, err := yaml.Marshal(env)
+	if err == nil {
+		return string(bs), nil
+	}
+
+	return "", err
+}
+
+func mapValuesPrimitive(value any) any {
+	if val, ok := value.(*Value); ok {
+		return mapValuesPrimitive(val.Value)
+	}
+	if mapData, isMap := value.(map[string]Value); isMap {
+
+		output := make(map[string]any, len(mapData))
+		for k, v := range mapData {
+			output[k] = mapValuesPrimitive(v.Value)
+		}
+
+		return output
+	} else if sliceData, isSlice := value.([]any); isSlice {
+		for i, v := range sliceData {
+			sliceData[i] = mapValuesPrimitive(v)
+		}
+		return sliceData
+	}
+
+	return value
```

Comment:
More idiomatic to express this as a type switch:

```suggestion
	switch val := value.(type) {
	case *Value:
		return mapValuesPrimitive(val.Value)
	case map[string]Value:
		output := make(map[string]any, len(mapData))
		for k, v := range mapData {
			output[k] = mapValuesPrimitive(v.Value)
		}
		return output
	case []any:
		for i, v := range sliceData {
			sliceData[i] = mapValuesPrimitive(v)
		}
		return sliceData
	default:
		return value
	}
```